_posts/2020-12-20-azure-event-grid.md

```diff
@@ -0,0 +1,55 @@
+- [Azure Event Hubs](https://azure.microsoft.com/en-us/services/event-hubs/?WT.mc_id=AZ-MVP-5001625) per la gestione degli stream di eventi
+- [Azure Event Grid](https://docs.microsoft.com/en-us/azure/event-grid/?WT.mc_id=AZ-MVP-5001625) per la gestione degli eventi di qualsiasi forma e origine
+- [Azure Service Bus](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview?WT.mc_id=AZ-MVP-5001625) per la gestione di scenari di business "critici" in cui si rivelano indispensabili le feature di un enterprise message broker completo
+
+Ora che abbiamo capito in che scenario ha senso pensare ad Event Grid, cerchiamo di capire un po' meglio cos'e': prima del suo avvento, tutti i servizi di Azure erano una serie di puntini, che potevano cooperare tra di loro, ma con modalita' completamente differenti tra di loro. Event Grid e' la lingua franca con cui mettere in comunicazione tutti questi servizi. La linea che permette di unire tutti i puntini.
+
+I [componenti principali](https://docs.microsoft.com/en-us/azure/event-grid/concepts?WT.mc_id=AZ-MVP-5001625) che caratterizzano Event Grid sono:
+- un evento, ovvero l'informazione che si vuol far "viaggiare" nel sistema
+- una sorgente (source), ovvero il componente che genera l'evento
+- un topic, ovvero l'endpoint a cui viene "inviato" l'evento
+- un handler, ovvero un componente interessato ad un particolare evento
+- una subscription, ovvero il collegamento tra un topic ed un componente interessato all'evento generato
+
+Nell'ecosistema di Azure, una _source_ puo' essere uno dei servizi esposti dalla piattaforma, come per esempio [Azure Container Registry](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-container-registry?WT.mc_id=AZ-MVP-5001625) o [Azure App Configuration](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-app-configuration?WT.mc_id=AZ-MVP-5001625). Un handler puo' essere per esempio una [Azure Function](https://docs.microsoft.com/en-us/azure/event-grid/handler-functions?WT.mc_id=AZ-MVP-5001625) o un [WebHook](https://docs.microsoft.com/en-us/azure/event-grid/handler-webhooks?WT.mc_id=AZ-MVP-5001625).  
+
+Visto che solitamente un'immagine vale piu' di mille parole, ecco tutti i servizi che e' possibile "collegare" sfruttando Event Grid:
+
+![functional model](https://docs.microsoft.com/en-us/azure/event-grid/media/overview/functional-model.png?WT.mc_id=AZ-MVP-5001625)
+
+Il fatto che uno dei possibili handler supportati da Event Grid sia il meccanismo dei [webhook](https://docs.microsoft.com/en-us/azure/event-grid/handler-webhooks?WT.mc_id=AZ-MVP-5001625) apre le porte all'integrazione di sistemi terzi (come per esempio nostre applicazioni) con le sorgenti supportate da Event Grid.
+
+Event Grid mette a disposizione la possibilita' di integrarsi anche dall'altro lato della barricata, ovvero dalla parte di chi produce gli eventi.  
```

Comment:
In che senso?

Reply:
Ho parlato dei webhook come possibile integrazione per il consumo di messaggi. In questo modo, oltre ai servizi di Azure (come per esempio le Function) anche componenti esterni alla piattaforma possono consumarli.

"l'altro lato della barricata" a questo punto (per me) e' chi gli eventi li produce. Anche in questo caso, oltre all'integrazione nativa per le sorgenti supportate da Azure, anche sorgenti esterne, tramite i _custom topic_ possono produrre eventi.

Piu' chiaro?

Reply:
Sì,direi di sì. Forse potresti rivedere il discorso introducendo l'idea che ci sia spazio per l'integrazione con produttori e consumatori di eventi esterni ed entrare poi nel dettaglio di come avvengono le due integrazioni (custom topic, web-hook). Che dici?